core/Util/plJPEG.cpp

```diff
@@ -195,7 +266,84 @@ void plJPEG::DecompressJPEG(hsStream* S, void* buf, size_t size)
     jpeg_finish_decompress(&ji.dinfo);
 }
 
-void plJPEG::CompressJPEG(hsStream* S, void* buf, size_t size)
+
+plMipmap* plJPEG::DecompressJPEG(hsStream* S)
+{
+    plJPEG& ji = Instance();
+
+    jpeg_hsStream_src(&ji.dinfo, S);
+    jpeg_read_header(&ji.dinfo, TRUE);
+    ji.dinfo.out_color_space = JCS_EXT_BGRA;    // Data stored as RGB on disk but Plasma uses BGR
+    jpeg_start_decompress(&ji.dinfo);
+
+    int row_stride = ji.dinfo.output_width * ji.dinfo.out_color_components;
+    int out_stride = row_stride;
+    RAII_JSAMPROW<1> jbuffer(row_stride);
+
+    // Start with a reasonable size for the buffer
+    std::vector<uint8_t> buffer(ji.dinfo.output_width * ji.dinfo.output_height * ji.dinfo.out_color_components);
```

Comment:
```suggestion
    auto buffer = std::make_unique<unsigned char[]>(ji.dinfo.output_width * ji.dinfo.output_height * ji.dinfo.out_color_components);
```

This is mostly a personal preference, but I'd prefer to see this as a `unique_ptr`. We should know the complete, uncompressed size of the buffer, so there really isn't any need for all of the features of a `vector`.